ocs_ci/ocs/hsbench.py

```diff
@@ -0,0 +1,192 @@
+import logging
+
+from tests import helpers
+from ocs_ci.utility import templating
+from ocs_ci.utility.utils import run_cmd
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.resources import pod
+from ocs_ci.framework import config
+from ocs_ci.ocs.exceptions import (
+    CommandFailed, UnexpectedBehaviour
+)
+
+log = logging.getLogger(__name__)
+
+
+class HsBench(object):
+    """
+    Hotsauce S3 benchmark
+
+    """
+
+    def __init__(self):
+        """
+        Initializer to treate pvc and rgw pod to running hsbench benchmark
+
+        Args:
+            kind (str): Kind of service POD or DeploymentConfig
+            pod_dict_path (yaml): Pod yaml
+            node_selector (dict): Pods will be created in this node_selector
+
+        """
+        self.pod_dic_path = constants.FEDORA_DC_YAML
+        self.namespace = config.ENV_DATA['cluster_namespace']
+        self.hsbench_cr = templating.load_yaml(constants.HSBENCH_OBJ_YAML)
+
+    def create_resource_hsbench(self):
+        """
+        Create resource for hsbench mark test:
+            Create service account
+            Create PVC
+            Create Fedora DC pod
+
+        """
+        # Check for existing rgw pods on cluster
+        self.rgw_pod = pod.get_rgw_pods()
+        if self.rgw_pod:
+            # Create service account
+            self.sa_name = helpers.create_serviceaccount(self.namespace)
+            self.sa_name = self.sa_name.name
+            helpers.add_scc_policy(
+                sa_name=self.sa_name, namespace=self.namespace
+            )
+
+            # Create test pvc+pod
+            log.info(f"Create Fedora dc pod for testing in {self.namespace}")
+            pvc_size = '50Gi'
+            node_name = 'compute-0'
+            self.pod_name = 'hsbench-pod'
+            self.pvc_obj = helpers.create_pvc(sc_name=constants.DEFAULT_STORAGECLASS_RBD,
+                                              namespace=self.namespace, size=pvc_size
+                                              )
+            self.pod_obj = helpers.create_pod(constants.CEPHBLOCKPOOL, namespace=self.namespace,
+                                              pod_name=self.pod_name, pvc_name=self.pvc_obj.name,
+                                              node_name=node_name, sa_name=self.sa_name,
+                                              pod_dict_path=self.pod_dic_path, dc_deployment=True,
+                                              deploy_pod_status=constants.STATUS_COMPLETED
+                                              )
+        else:
+            raise UnexpectedBehaviour("This cluster doesn't have RGW pod(s) to perform hsbench")
```

Comment:
Why only RGW? Can this be generalised to work on Noobaa, AWS and other s3 platforms?

Reply:
This test case is created to cover customer issue: https://bugzilla.redhat.com/show_bug.cgi?id=1809545.
It can be enhanced to run with aws and others.

Reply:
Presumably historical.

Reply:
Okay